mlflow/openai/__init__.py

```diff
@@ -163,6 +164,30 @@ def _get_task_name(task):
         )
 
 
+def _get_api_config() -> Dict[str, Any]:
+    """
+    Gets the parameters and configuration of the OpenAI API connected to.
+    """
+    import openai
+
+    config = {}
+    api_type = os.getenv(_OpenAIEnvVar.OPENAI_API_TYPE.value, openai.api_type)
+    if api_type in ("azure", "azure_ad", "azuread"):
+        config["batch_size"] = 16
+        config["max_requests_per_minute"] = 3_500
+        config["max_tokens_per_minute"] = 60_000
+    else:
+        # The maximum batch size is 2048:
+        # https://github.com/openai/openai-python/blob/b82a3f7e4c462a8a10fa445193301a3cefef9a4a/openai/embeddings_utils.py#L43
+        # We use a smaller batch size to be safe.
+        config["batch_size"] = 1024
+        config["max_requests_per_minute"] = 3_500
+        config["max_tokens_per_minute"] = 90_000
+    config["api_type"] = api_type
```

Comment:
nit: could we define this as part of the original constructor in line 173, relocated below line 174?

```python
    api_type = os.getenv(_OpenAIEnvVar.OPENAI_API_TYPE.value, openai.api_type)
    config = {"api_type": api_type}